internal/users/manager.go

```diff
@@ -213,6 +211,21 @@ func (m *Manager) UpdateUser(u types.UserInfo) (err error) {
 	return nil
 }
 
+func (m *Manager) findGroup(group types.GroupInfo) (oldGroup cache.GroupDB, err error) {
+	// Search by UGID first to support renaming groups
+	oldGroup, err = m.cache.GroupByUGID(group.UGID)
+	if err != nil && !errors.Is(err, cache.NoDataFoundError{}) {
+		return oldGroup, err
+	}
+	if err == nil {
+		return oldGroup, nil
+	}
```

Comment:
Maybe it reads better as

```suggestion
	if err == nil {
		return oldGroup, nil
	}
	if !errors.Is(err, cache.NoDataFoundError{}) {
		return oldGroup, err
	}
```

Reply:
done